File: problems/1248_count_number_of_nice_subarrays/Solution.java
import java.util.*;

class Solution {
    public int numberOfSubarrays(int[] nums, int k) {
        Map<Integer, Integer> freq = new HashMap<>();
        freq.put(0, 1);
        int prefix = 0, count = 0;
        for (int x : nums) {
            prefix += x % 2;
            count += freq.getOrDefault(prefix - k, 0);
            freq.merge(prefix, 1, Integer::sum);
        }
        return count;
    }
}

File: problems/1248_count_number_of_nice_subarrays/solution.py
class Solution:
    def numberOfSubarrays(self, nums: list[int], k: int) -> int:
        count = 0
        prefix = 0
        freq = {0: 1}
        for x in nums:
            prefix += x % 2
            count += freq.get(prefix - k, 0)
            freq[prefix] = freq.get(prefix, 0) + 1
        return count
